### PR TITLE
Refactor/20240409/moves qub docs to ulisboa

### PR DESCRIPTION
**Pull request by joaofreitas-qubit**

### CODE REVIEW

File: src/main/java/org/fenixedu/treasury/domain/paypal/PayPal.java
Comment (pabrantes) [inline src/main/java/org/fenixedu/treasury/domain/paypal/PayPal.java:186]: @anilmamede although we do not expect any suprises in this change to use GSON. Can you make sure we get this part validated? Thank you!

File: src/main/java/org/fenixedu/treasury/services/reports/DocumentPrinter.java
Comment (pabrantes) [inline src/main/java/org/fenixedu/treasury/services/reports/DocumentPrinter.java:73]: @joaofreitas-qubit we can simplify the code having this part refactored:

`        
 private static <T extends Object> T apply(Function<Xpto, T> function) {
        return Optional.ofNullable(SINGLETON).map(function).orElseThrow(() -> new RuntimeException("Feature not available"));
    }
`

And then you can simply do in the methods:

`
 public static byte[] printDebtAccountPaymentPlan(DebtAccount debtAccount, String outputMimeType) {
   apply(dp -> dp.printDebtAccountPaymentPlan(debtAccount,outputMimeType));
 }
`
Although this can reduce some code, I'm not blocking the PR due to this refactor.

File: src/main/java/org/fenixedu/treasury/util/TreasuryBootstrapper.java
Comment (pabrantes) [inline src/main/java/org/fenixedu/treasury/util/TreasuryBootstrapper.java:117]: @joaofreitas-qubit we can remove this change, since it's just a formatting issue

**Review by pabrantes — APPROVED** [review]
Although I've done a few comments nothing is really blocking so I'm approving this. We still need to get the green light from @anilmamede before proceeding with this merge

**Review by anilmamede — APPROVED** [review]
I could not test but I got to the point of printing but I lack soffice in my dev env . I reviewed the code and and it fine

**Pull request merged**
